client/app/d3Dualbars/d3Dualbars.legend.directive.js

```diff
@@ -0,0 +1,187 @@
+    return {
+      restrict: 'EA',
+      scope: {},
+      link: function (scope, element, attrs) {
+
+      	var barHeight = parseInt(attrs.barHeight) || 40;
+        var barPadding = parseInt(attrs.barPadding) || 5;
+        var graphMargin = parseInt(attrs.graphMargin) || 30;
+        var barMargin = parseInt(attrs.barMargin) || 5;
+        var textMargin = parseInt(attrs.textMargin) || 5;
+        var groupPadding = parseInt(attrs.groupPadding) || 60;
+        var topPadding = 20;
+        var sidePadding = 100;
+        var defaultConfig = {
+          barOpacity: 0.05,
+          upperRadius: 0.45,
+          autoBarHeight: true
+        };
+        // init svg
+        var svg = d3.select(element[0])
+          .append('svg')
+          .style('width', '250%')
+
+        var timeout;
+        $window.addEventListener('resize', function () {
+          if (timeout) $timeout.cancel(timeout);
+          timeout = $timeout(function () {
+            scope.render(preprocess(scope.data), scope.config);
+          }, 500);
+        });
+
+        var width = d3.select(element[0]).node().offsetWidth - graphMargin - groupPadding - sidePadding;
+        console.log(width);
+
+        var radiusScale = d3.scale.linear()
+            .domain([0, .45])
+            .range([3, 57/2]);
+
+  		  var legendY = 70;
+          var legendX= 600;
+          var legend = svg.append('g')
+            .attr('class', 'legend')
+            .attr('x', function () {})
+            .attr('transform', 'translate(' + legendX + ',' + legendY + ')')
+            .attr('height', 100)
+            .attr('width', 100);
+
+          var relative = legend.append('g')
+            .append('g')
+            .attr('transform', 'translate(-205, -60)'); //'translate(' + (width - 889) + ',' + -60 +')'); when width = 684
+          relative.append('line')
+            .attr('x1', -50)
+            .attr('y1', 20)
+            .attr('x2', 30)
+            .attr('y2', 20)
+            .style('stroke-width', 1)
+            .style('stroke-dasharray', ('3, 3'))
+            .style('stroke', '#fff')
+            .style('fill', 'none');
+          relative.append('text')
+            .attr('x', -175)
+            .attr('y', 25)
+            .attr('height', 30)
+            .attr('width', 100)
+            .attr('font-size', '.8em')
+            .text('Percentile Risk:')
+            .style('fill','#fff');
+
+          var absolute = legend.append('g')
+            .append('g')
+            .attr('transform', 'translate(-380, 5)'); //'translate(' + (width-1064) + ',' + 5 + ')');
+
+          absolute.append('text')
+            .attr('font-size', '.8em')
+            .text('Absolute Risk:')
+            .style('fill','#fff');
+
+          var twoFive = legend.append('g')
+            .append('g')
+            .attr('transform', 'translate(-110, 0)'); //'translate(' + (width-794) + ',' + 0 + ')');
+          twoFive.append('circle')
```

Comment:
Very likely there is a better way to make the circles. However, for this PR we will keep it this way.